addons/content-index/src/main/java/org/commonjava/indy/content/index/DefaultContentIndexManager.java

```diff
@@ -245,21 +246,41 @@ public void clearAllIndexedPathInStore( ArtifactStore store )
     @Override
     public void clearAllIndexedPathWithOriginalStore( ArtifactStore originalStore )
     {
+        StoreKey osk = originalStore.getKey();
+
         Set<IndexedStorePath> isps =
-                contentIndex.cacheKeySetByFilter( key -> key.getOriginStoreKey().equals( originalStore.getKey() ) );
+                contentIndex.cacheKeySetByFilter( (key) -> {
+                    StoreKey myOriginKey = key.getOriginStoreKey();
+                    return myOriginKey != null && myOriginKey.equals( osk );
+                } );
+
         if ( isps != null )
         {
             isps.forEach( isp -> contentIndex.remove( isp ) );
         }
+
         logger.trace( "Clear all indices with origin: {}, size: {}", originalStore.getKey(), ( isps != null ? isps.size() : 0 ) );
     }
 
     @Override
     public void clearAllIndexedPathInStoreWithOriginal( ArtifactStore store, ArtifactStore originalStore )
```

Comment:
Does this enhance the above one for better performance? (clearAllIndexedPathWithOriginalStore) 
Or do we have a use-case that needs to clean for an individual store with certain origin? 
If it is for performance, I guess ISPN may just do a linear search in both methods and this may not be very useful.